services/src/datasets/external/gbif.rs

```diff
@@ -384,10 +795,154 @@ impl LayerCollectionProvider for GbifDataProvider {
         })
     }
-}
 
-#[async_trait]
-impl
+    async fn get_search_capabilities(&self) -> Result<SearchCapabilities> {
+        Ok(SearchCapabilities {
+            search_types: SearchTypes {
+                fulltext: true,
+                prefix: true,
+            },
+            autocomplete: true,
+            filters: None,
+        })
+    }
+
+    async fn search(
+        &self,
+        collection_id: &LayerCollectionId,
+        search: SearchParameters,
+    ) -> Result<LayerCollection> {
+        let selector = collection_id.0.split_once('/').map_or_else(
+            || collection_id.clone().0,
+            |(selector, _)| selector.to_string(),
+        );
+        let path = collection_id
+            .0
+            .split_once('/')
+            .map_or_else(|| "", |(_, path)| path);
+
+        let search_string = match search.search_type {
+            SearchType::Fulltext => format!("%{}%", search.search_string),
+            SearchType::Prefix => format!("{}%", search.search_string),
+        };
+
+        let items = match selector.as_str() {
+            "datasets" => {
+                let taxonrank = path
+                    .split_once('/')
+                    .map_or_else(String::new, |(taxonrank, _)| taxonrank.to_string());
+                ensure!(
+                    Self::LISTABLE_RANKS.contains(&taxonrank.as_str()),
+                    crate::error::InvalidPath
+                );
+                self.get_all_datasets_search_items(
+                    &search.limit,
+                    &search.offset,
+                    path,
+                    &search_string,
+                    vec![taxonrank],
+                )
+                .await?
+            }
+            "filter" => {
+                self.get_filter_search_items(&search.limit, &search.offset, path, &search_string)
+                    .await?
+            }
+            "select" => {
+                self.get_all_datasets_search_items(
+                    &search.limit,
+                    &search.offset,
+                    path,
+                    &search_string,
+                    Self::LISTABLE_RANKS
+                        .map(std::string::ToString::to_string)
+                        .to_vec(),
+                )
+                .await?
+            }
+            _ => vec![],
+        };
+
+        Ok(LayerCollection {
+            id: ProviderLayerCollectionId {
+                provider_id: GBIF_PROVIDER_ID,
+                collection_id: collection_id.clone(),
+            },
+            name: "GBIF".to_owned(),
+            description: "GBIF occurrence datasets".to_owned(),
+            items,
+            entry_label: None,
+            properties: vec![],
+        })
+    }
+
+    async fn autocomplete_search(
+        &self,
+        collection_id: &LayerCollectionId,
+        search: SearchParameters,
+    ) -> Result<Vec<String>> {
+        let selector = collection_id.0.split_once('/').map_or_else(
```

Comment:
so a search term cannot contain "/"? Shouldn't we escape that then?

Reply:
Aren't the query parameters already stripped from the path at this point by the handler?

Reply:
yes, but why do you split at "/" here?

Reply:
It's the way the collection-/layer-IDs are structured for the GBIF provider. In this instance, the selector datasets/filter/select determines whether you want to search within the family/genus/species datasets list, the options for applying a filter along the taxonomic "path" or perform a search across family, genus and species datasets